rosdep/base.yaml

```diff
@@ -546,9 +546,6 @@ clang-tidy:
   fedora: [clang-tools-extra]
   gentoo: [sys-devel/clang]
   nixos: [clang]
-  osx:
-    homebrew:
-      packages: [llvm]
```

Comment:
why is this deleted?

IIUC, llvm includes clang-tidy when installed

Reply:
> So to make clang-tidy work, we need to manually put it to $PATH.

Which one? The one provided by Xcode?
If that's the case, removing the rule sounds fine, but if installing `llvm` through homebrew is still needed I wouldn't delete the rule.

Reply:
@ivanpauno I mean the homebrew one. Xcode does not include a ```tidy-clang```. But even after installing ```llvm``` from homebrew, ```rosdep``` still fails because it query the installation with 
```
brew info llvm --json=v1
```
and the check the ```linked_keg``` in the output. For ```keg-only``` packages, ```linked_keg``` key is null so ```rosdep``` fails. You can take a look at this code:
https://github.com/ros-infrastructure/rosdep/blob/9f5829869db6c28d3cb0a536c699b82919a4c358/src/rosdep2/platforms/osx.py#L207

It seems currently there is no easy way to automatically install ```clang-tidy```. As long as we install it with homebrew (either by ```rosdep``` or manually), we have to manually link the executable to ```$PATH```. But removing ```llvm``` from ```base.yaml``` will at least make ```rosdep``` happy and maybe instructions should be added to documentations.

Reply:
that sounds fair enough to me